// SPDX-FileCopyrightText: 2025 Carnegie Mellon University - Living Edge Lab
//
// SPDX-License-Identifier: GPL-2.0-only

/**
 * Messages/calls related to the Gabriel component of SteelEagle.
 *
 * The gabriel extras is a specialized protocol just for intergrating with
 * gabriel compute services.
 */
syntax = "proto3";

package protocol.gabriel_extras;
import "dataplane.proto";
import "controlplane.proto";

message Extras {
  // Designates an initial request to register a drone
  bool registering = 1;
  // The telemetry data for the drone
  protocol.dataplane.Telemetry telemetry = 2;
  // Configuration parameters for the compute request
<<<<<<< HEAD
  protocol.controlplane.Request cpt_request = 4;
  protocol.controlplane.Response cpt_response = 5;
=======
  protocol.controlplane.Request cpt_request = 3;
  // Used for sending a reply back to the client
  protocol.controlplane.Response cpt_reply = 4;
>>>>>>> origin/main
}
